### ISSUE TITLE
kickasstorrent-kathow - Error while parsing row

### ISSUE THREAD

**Issue by Jorman** [Jackett/Jackett#3072]
Hi, I got this error on Jackett logs

I tried to fix it by myself but no lucky. If anyone is able to fix it, can explain to me how to do that?
I want to help when I can, but I'm feel quite stupid ... I can't help even in this simple fix

here the log:
```
CardigannIndexer (kickasstorrent-kathow): Error while parsing row '<tr class="odd" id="torrent_latest_torrents"> <td> <div class="iaconbox center floatright"> <a class="icommentjs kaButton smallButton rightButton">0<i class="ka ka-comment">				</i>
			</a> <a rel="nofollow, noreferrer" data-nop="" title="Torrent magnet link" href="https://mylink.bz/?url=magnet%3A%3Fxt%3Durn%3Abtih%3AE6D211E53AAF28221A4281F141C21F9AEF941E8C%26dn%3DThe%2BBig%2BBang%2BTheory%2BS11E16%2B2018%2BHDTV%2Bx264%2B-%2BCM%26tr%3Dudp%253A%252F%252Ftracker.openbittorrent.com%253A80%26tr%3Dudp%253A%252F%252Fopen.demonii.com%253A1337%26tr%3Dudp%253A%252F%252Ftracker.coppersurfer.tk%253A6969%26tr%3Dudp%253A%252F%252Ftracker.opentrackr.org%253A1337%252Fannounce" class="icon16">
				<i class="ka ka16 ka-magnet">				</i>
			</a> <a rel="nofollow, noreferrer" data-download="" title="Download torrent file" href="https://mylink.bz/?url=magnet%3A%3Fxt%3Durn%3Abtih%3AE6D211E53AAF28221A4281F141C21F9AEF941E8C%26dn%3DThe%2BBig%2BBang%2BTheory%2BS11E16%2B2018%2BHDTV%2Bx264%2B-%2BCM%26tr%3Dudp%253A%252F%252Ftracker.openbittorrent.com%253A80%26tr%3Dudp%253A%252F%252Fopen.demonii.com%253A1337%26tr%3Dudp%253A%252F%252Ftracker.coppersurfer.tk%253A6969%26tr%3Dudp%253A%252F%252Ftracker.opentrackr.org%253A1337%252Fannounce" target="_blank" class="icon16">
				<i class="ka ka16 ka-arrow-down">				</i>
			</a> <!--<a data-download title="Download torrent file" href="http://itorrents.org/torrent/e6d211e53aaf28221a4281f141c21f9aef941e8c.torrent?title=%5Bkat.li%5DThe+Big+Bang+Theory+S11E16+2018+HDTV+x264+-+CM" target="_blank" class="icon16"><i class="ka ka16 ka-arrow-down"></i></a>--> </div> <div class="torrentname"> <a href="/t2-The-Big-Bang-Theory-S11E16-2018-HDTV-x264-CM.html" class="torType film">			</a> <a href="/t2-The-Big-Bang-Theory-S11E16-2018-HDTV-x264-CM.html" class="normalgrey font12px plain bold">			</a> <div class="markeredBlock torType filmType"> <a href="/t2-The-Big-Bang-Theory-S11E16-2018-HDTV-x264-CM.html" class="cellMainLink">The Big Bang Theory S11E16 2018 HDTV x264 - CM</a> <span class="font11px lightgrey block">    </span> </div> </div>
	</td> <td class="nobr center">324.34 MB</td> <td class="center" title="2 months ago">2 months ago</td> <td class="green center">N/A</td> <td class="red lasttd center">N/A</td> </tr>':

System.Exception: Error while parsing field=category, selector=span[id^="cat_"] > strong > a, value=<null>: Selector "span[id^="cat_"] > strong > a" didn't match <tr class="odd" id="torrent_latest_torrents"> <td> <div class="iaconbox center floatright"> <a class="icommentjs kaButton smallButton rightButton">0<i class="ka ka-comment">				</i>
			</a> <a rel="nofollow, noreferrer" data-nop="" title="Torrent magnet link" href="https://mylink.bz/?url=magnet%3A%3Fxt%3Durn%3Abtih%3AE6D211E53AAF28221A4281F141C21F9AEF941E8C%26dn%3DThe%2BBig%2BBang%2BTheory%2BS11E16%2B2018%2BHDTV%2Bx264%2B-%2BCM%26tr%3Dudp%253A%252F%252Ftracker.openbittorrent.com%253A80%26tr%3Dudp%253A%252F%252Fopen.demonii.com%253A1337%26tr%3Dudp%253A%252F%252Ftracker.coppersurfer.tk%253A6969%26tr%3Dudp%253A%252F%252Ftracker.opentrackr.org%253A1337%252Fannounce" class="icon16">
				<i class="ka ka16 ka-magnet">				</i>
			</a> <a rel="nofollow, noreferrer" data-download="" title="Download torrent file" href="https://mylink.bz/?url=magnet%3A%3Fxt%3Durn%3Abtih%3AE6D211E53AAF28221A4281F141C21F9AEF941E8C%26dn%3DThe%2BBig%2BBang%2BTheory%2BS11E16%2B2018%2BHDTV%2Bx264%2B-%2BCM%26tr%3Dudp%253A%252F%252Ftracker.openbittorrent.com%253A80%26tr%3Dudp%253A%252F%252Fopen.demonii.com%253A1337%26tr%3Dudp%253A%252F%252Ftracker.coppersurfer.tk%253A6969%26tr%3Dudp%253A%252F%252Ftracker.opentrackr.org%253A1337%252Fannounce" target="_blank" class="icon16">
				<i class="ka ka16 ka-arrow-down">				</i>
			</a> <!--<a data-download title="Download torrent file" href="http://itorrents.org/torrent/e6d211e53aaf28221a4281f141c21f9aef941e8c.torrent?title=%5Bkat.li%5DThe+Big+Bang+Theory+S11E16+2018+HDTV+x264+-+CM" target="_blank" class="icon16"><i class="ka ka16 ka-arrow-down"></i></a>--> </div> <div class="torrentname"> <a href="/t2-The-Big-Bang-Theory-S11E16-2018-HDTV-x264-CM.html" class="torType film">			</a> <a href="/t2-The-Big-Bang-Theory-S11E16-2018-HDTV-x264-CM.html" class="normalgrey font12px plain bold">			</a> <div class="markeredBlock torType filmType"> <a href="/t2-The-Big-Bang-Theory-S11E16-2018-HDTV-x264-CM.html" class="cellMainLink">The Big Bang Theory S11E16 2018 HDTV x264 - CM</a> <span class="font11px lightgrey block">    </span> </div> </div>
	</td> <td class="nobr center">324.34 MB</td> <td class="center" title="2 months ago">2 months ago</td> <td class="green center">N/A</td> <td class="red lasttd center">N/A</td> </tr>
  at Jackett.Common.Indexers.CardigannIndexer+<PerformQuery>d__29.MoveNext () [0x0163d] in C:\projects\jackett\src\Jackett.Common\Indexers\CardigannIndexer.cs:1450 
```

**Comment by garfield69** [comment]
so the issue here is that you came across a torrent that was uploaded to the site without a category.
so to prevent future errors I've made the category selector *optional*.
the next (pre)release of Jackett should include fix fb316d9068def21ad254bc018c420fc364a56861

**Comment by Jorman** [comment]
Till now, I don't know how you can do this, you make it like if was simple ... Always thanks!

I think now there's a kind of joke, now the odd clas is fixed, but see here
```

CardigannIndexer (kickasstorrent-kathow): Error while parsing row '<tr class="even" id="torrent__torrents"> <td> <div class="iaconbox center floatright"> <a class="icommentjs kaButton smallButton rightButton">0<i class="ka ka-comment">				</i>
			</a> <a rel="nofollow, noreferrer" data-nop="" title="Torrent magnet link" href="https://mylink.bz/?url=magnet%3A%3Fxt%3Durn%3Abtih%3AD5B2AC481898B57328A3DEE35344A63EBF97E1DC%26dn%3DRaazi%2B%25282018%2529%2BHindi%2BNew%2B720p%2BPre-DvDRip%2Bx264%2BAAC%2B1GB%255BMRbuZ%255D%26tr%3Dudp%253A%252F%252Ftracker.openbittorrent.com%253A80%252Fannounce%26tr%3Dudp%253A%252F%252Ftracker.publicbt.com%253A80%252Fannounce%26tr%3Dudp%253A%252F%252Ftracker.torrent.eu.org%253A451%252Fannounce%26tr%3Dhttp%253A%252F%252Ft.nyaatracker.com%253A80%252Fannounce%26tr%3Dudp%253A%252F%252Ftracker.doko.moe%253A6969%252Fannounce%26tr%3Dhttp%253A%252F%252Fasnet.pw%253A2710%252Fannounce%26tr%3Dudp%253A%252F%252Fthetracker.org%253A80%252Fannounce%26tr%3Dhttp%253A%252F%252Ftracker.tfile.co%253A80%252Fannounce%26tr%3Dhttp%253A%252F%252Fpt.lax.mx%253A80%252Fannounce%26tr%3Dudp%253A%252F%252Fsantost12.xyz%253A6969%252Fannounce%26tr%3Dhttps%253A%252F%252Ftracker.bt-hash.com%253A443%252Fannounce%26tr%3Dudp%253A%252F%252Fbt.xxx-tracker.com%253A2710%252Fannounce%26tr%3Dudp%253A%252F%252Ftracker.vanitycore.co%253A6969%252Fannounce%26tr%3Dudp%253A%252F%252Fzephir.monocul.us%253A6969%252Fannounce%26tr%3Dudp%253A%252F%252Ftracker.zer0day.to%253A1337%252Fannounce%26tr%3Dudp%253A%252F%252Ftracker.leechers-paradise.org%253A6969%252Fannounce%26tr%3Dudp%253A%252F%252Fcoppersurfer.tk%253A6969%252Fannounce" class="icon16">
				<i class="ka ka16 ka-magnet">				</i>
			</a> <a rel="nofollow, noreferrer" data-download="" title="Download torrent file" href="https://mylink.bz/?url=magnet%3A%3Fxt%3Durn%3Abtih%3AD5B2AC481898B57328A3DEE35344A63EBF97E1DC%26dn%3DRaazi%2B%25282018%2529%2BHindi%2BNew%2B720p%2BPre-DvDRip%2Bx264%2BAAC%2B1GB%255BMRbuZ%255D%26tr%3Dudp%253A%252F%252Ftracker.openbittorrent.com%253A80%252Fannounce%26tr%3Dudp%253A%252F%252Ftracker.publicbt.com%253A80%252Fannounce%26tr%3Dudp%253A%252F%252Ftracker.torrent.eu.org%253A451%252Fannounce%26tr%3Dhttp%253A%252F%252Ft.nyaatracker.com%253A80%252Fannounce%26tr%3Dudp%253A%252F%252Ftracker.doko.moe%253A6969%252Fannounce%26tr%3Dhttp%253A%252F%252Fasnet.pw%253A2710%252Fannounce%26tr%3Dudp%253A%252F%252Fthetracker.org%253A80%252Fannounce%26tr%3Dhttp%253A%252F%252Ftracker.tfile.co%253A80%252Fannounce%26tr%3Dhttp%253A%252F%252Fpt.lax.mx%253A80%252Fannounce%26tr%3Dudp%253A%252F%252Fsantost12.xyz%253A6969%252Fannounce%26tr%3Dhttps%253A%252F%252Ftracker.bt-hash.com%253A443%252Fannounce%26tr%3Dudp%253A%252F%252Fbt.xxx-tracker.com%253A2710%252Fannounce%26tr%3Dudp%253A%252F%252Ftracker.vanitycore.co%253A6969%252Fannounce%26tr%3Dudp%253A%252F%252Fzephir.monocul.us%253A6969%252Fannounce%26tr%3Dudp%253A%252F%252Ftracker.zer0day.to%253A1337%252Fannounce%26tr%3Dudp%253A%252F%252Ftracker.leechers-paradise.org%253A6969%252Fannounce%26tr%3Dudp%253A%252F%252Fcoppersurfer.tk%253A6969%252Fannounce" class="icon16">
				<i class="ka ka16 ka-arrow-down">				</i>
			</a> </div> <div class="torrentname"> <a href="/t3-Raazi-2018-Hindi-New-720p-Pre-DvDRip-x264-AAC-1GB-MRbuZ-tt2989383.html" class="torType VideoType">			</a> <a href="/t3-Raazi-2018-Hindi-New-720p-Pre-DvDRip-x264-AAC-1GB-MRbuZ-tt2989383.html" class="normalgrey font12px plain bold">			</a> <div class="markeredBlock torType VideoType"> <a href="/t3-Raazi-2018-Hindi-New-720p-Pre-DvDRip-x264-AAC-1GB-MRbuZ-tt2989383.html" class="cellMainLink">Raazi (2018) Hindi New 720p Pre-DvDRip x264 AAC 1GB[MRbuZ]</a> <span class="font11px lightgrey block">  Posted by  ictv  in <span id="cat_12975568">
						<strong>
							<a href="/movies" target="_blank">Movies</a>
						</strong>
					</span>  </span>
			</div> </div>
	</td> <td class="nobr center">1.015.9 MB</td> <td class="center" 6="" hours="" ago"="">6 hours ago</td> <td class="green center">30</td> <td class="red lasttd center"> 44</td> </tr>':

System.Exception: Error while parsing field=size, selector=td:nth-child(2), value=1.015.9 MB: Input string was not in a correct format.
  at Jackett.Common.Indexers.CardigannIndexer+<PerformQuery>d__29.MoveNext () [0x0163d] in C:\projects\jackett\src\Jackett.Common\Indexers\CardigannIndexer.cs:1450
```

Now is time of even row!

What do you think?

**Comment by kaso17** [comment]
@Jorman that error was caused by an invalid size format (1.015.9 MB)
fixed via bc34b9f1765e94f0c79c44d6eecd34c7d7407747
